airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/streams/common.py

```diff
@@ -150,7 +152,7 @@ def traced_exception(fb_exception: FacebookRequestError):
     Please see ../unit_tests/test_errors.py for full error examples
     Please add new errors to the tests
     """
-    msg = fb_exception.api_error_message()
+    msg = fb_exception.api_error_message() if fb_exception.api_error_message() else fb_exception.get_message()
```

Comment:
Why is this assignment here instead of at L#193, such as:
```
elif not fb_exception.api_error_message() and fb_exception.http_status() == 503:
    msg = fb_exception.get_message()
```
I am mentioning that just in case this assignment here might affect existing validations over msg content.

Reply:
Thanks for the review! The failure we were encountering is being thrown on the next line (L# 157):

```
if "Error validating access token" in msg:
```

This is currently resulting in a `TypeError` whenever msg evaluates to `None`, so any errors that don't contain `api_error_message` will get caught up here and not constructed as `AirbyteTracedExceptions` (not sure how widespread it is, but there's at least one other unrelated error that I noticed failing on the same location). By enforcing a value for `msg` here, I'm hoping to add a safety net for any errors that fall under this scenario, while still maintaining the existing assignment for errors that do have the `api_error_message`. Let me know if that addresses your concern!

Reply:
Can I suggest:
```suggestion
    msg = fb_exception.api_error_message() or fb_exception.get_message()
```

Also for line 157 have you tried?
```suggestion
   # probably "if msg" would also work as truly check, but some people prefer the full not None check
   if msg is not None and "Error validating access token" in msg:
```

I believe this way you could move the other evaluation to L#193 as Cristina suggested.